meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/DistributedSemaphore.java

```diff
@@ -50,23 +87,50 @@ public DistributedSemaphore(MetaClientInterface<DataRecord> client) {
    * @param capacity capacity of the semaphore
    */
   public void createSemaphore(String path, int capacity) {
-    throw new NotImplementedException("Not implemented yet.");
+    if (capacity <= 0) {
+      throw new MetaClientException("Capacity must be positive");
+    }
+    if (path == null || path.isEmpty()) {
+      throw new MetaClientException("Invalid path to create semaphore");
+    }
+    if (_metaClient.exists(path) != null) {
+      throw new MetaClientException("Semaphore already exists");
+    }
+    _path = path;
```

Comment:
Shouldn't we do this after line 104? What if creation of znode fails?

Reply:
Correct I'll fix that, thanks!